### PR TITLE
Simplifying experiments

### PR DESCRIPTION
**Pull request by witwolf**

### CODE REVIEW

File: alf/examples/actor_critic.py [hunk 1]
```diff
@@ -1,229 +0,0 @@
...
-python actor_critic.py \
-  --root_dir=~/tmp/socialbot-cartpole \
-  --alsologtostderr \
-  --gin_param='create_environment.env_name="SocialBot-CartPole-v0"' \
-  --gin_param='create_environment.num_parallel_environments=16' \
-  --gin_param='create_environment.env_load_fn=@suite_socialbot.load' \
-  --gin_file=ac_cart_pole.gin
-```
-
-To Run on SocailBot SimpleNavigation
```
Comment (Jialn) [inline alf/examples/actor_critic.py]: Maybe add these removed examples of running command to main.py? Sometimes they are useful.

File: alf/algorithms/merlin_algorithm_test.py [hunk 2]
```diff
@@ -22,61 +21,17 @@
...
-            rl_loss=ActorCriticLoss(action_spec=action_spec, gamma=1.0),
-            train_step_counter=global_step,
-            optimizer=optimizer,
-            debug_summaries=True)
-
-        return algorithm
-
     def _create_policy(self, env, train_interval=1, learning_rate=1e-1):
-        algorithm = self._create_algorithm(env, learning_rate)
+        algorithm = create_merlin_algorithm(env, learning_rate)
```
Comment (emailweixu) [inline alf/algorithms/merlin_algorithm_test.py]: TrainingPolicy can be removed

File: alf/environments/utils.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        py_env = parallel_py_environment.ParallelPyEnvironment(
+            [lambda: env_load_fn(env_name)] * num_parallel_environments)
+    return tf_py_environment.TFPyEnvironment(py_env)
+
+
+@gin.configurable
+def load_with_random_max_episode_steps(env_name,
+                                       env_load_fn=suite_gym.load,
+                                       min_steps=200,
+                                       max_steps=250):
```
Comment (emailweixu) [inline alf/environments/utils.py]: need comment

File: alf/examples/main.py [hunk 4]
```diff
@@ -77,6 +77,7 @@ def train_eval(train_dir,
         debug_summaries=debug_summaries)
 
 
+@gin.configurable
```
Comment (emailweixu) [inline alf/examples/main.py]: Is it possible to avoid having to specify both play.algorithm_ctor train_eval.algorithm_ctor?

File: alf/algorithms/actor_critic_algorithm_test.py [hunk 5]
```diff
@@ -61,38 +62,17 @@ def _create_policy(self,
...
-            loss=ActorCriticLoss(action_spec=action_spec, gamma=1.0),
-            optimizer=optimizer)
+        algorithm = create_ac_algorithm(
+            env=env,
+            actor_fc_layers=(),
+            value_fc_layers=(),
+            learning_rate=learning_rate,
+            use_rnns=use_rnn)
 
         policy = TrainingPolicy(
```
[the remark this answers is not in the document]
Comment (witwolf) [reply alf/algorithms/actor_critic_algorithm_test.py]: test already existed in on_policy_driver_test ,  so removed this file

File: alf/examples/main.py [hunk 6]
```diff
@@ -11,56 +11,77 @@
...
 
 FLAGS = flags.FLAGS
 
 
 @gin.configurable
-def train_eval(train_dir, evaluate=True, debug_summaries=False):
-    """A simple train and eval for ActorCriticAlgorithm."""
+def train_eval(train_dir,
+               algorithm_ctor=create_ac_algorithm,
+               trainer=on_policy_trainer.train,
```
[the remark this answers is not in the document]
Comment (witwolf) [reply alf/examples/main.py]: updated

File: alf/drivers/off_policy_driver_test.py [hunk 7]
```diff
@@ -12,110 +12,55 @@
...
+        actor_learning_rate=1e-2,
+        critic_learning_rate=1e-1)
 
-    @parameterized.parameters(create_sac_algorithm, create_ddpg_algorithm, )
+
+class OffPolicyDriverTest(parameterized.TestCase, unittest.TestCase):
+    def setUp(self) -> None:
+        gin.parse_config([
+            "ActorDistributionRnnNetwork.lstm_size=(4,)",
+            "CriticRnnNetwork.lstm_size=(4,)"
```
[the remark this answers is not in the document]
Comment (witwolf) [reply alf/drivers/off_policy_driver_test.py]: removed

File: alf/algorithms/merlin_algorithm_test.py (hunk 2 above)
Comment (witwolf) [reply alf/algorithms/merlin_algorithm_test.py]: removed

File: alf/environments/utils.py (hunk 3 above)
Comment (witwolf) [reply alf/environments/utils.py]: updated

File: alf/examples/main.py (hunk 4 above)
Comment (witwolf) [reply alf/examples/main.py]: bind `__main__.play.algorithm_ctor ` with `__main__.train_eval.algorithm_ctor`

File: alf/examples/actor_critic.py (hunk 1 above)
Comment (witwolf) [reply alf/examples/actor_critic.py]: All examples run in the same way
`python main.py --root_dir=${root_dir} --gin_file=${gin_flie} [--gin_param=...]`

File: alf/examples/main.py [hunk 8]
```diff
@@ -69,6 +69,10 @@ def train_eval(train_dir,
         eval_env = None
     algorithm = algorithm_ctor(env, debug_summaries=debug_summaries)
 
+    if isinstance(algorithm, OffPolicyAlgorithm):
+        trainer = off_policy_trainer.train
+    else:
```
Comment (emailweixu) [inline alf/examples/main.py]: else isinstance( ..., OnPolicyAlgorithm)
else:
   error

File: alf/examples/main.py [hunk 9]
```diff
@@ -11,56 +11,81 @@
...
 FLAGS = flags.FLAGS
 
 
 @gin.configurable
-def train_eval(train_dir, evaluate=True, debug_summaries=False):
-    """A simple train and eval for ActorCriticAlgorithm."""
+def train_eval(train_dir,
+               algorithm_ctor=create_ac_algorithm,
+               evaluate=True,
+               debug_summaries=False):
```
Comment (emailweixu) [inline alf/examples/main.py]: comment this function. And also note that play.algorithm_ctor is bind with this algorithm_ctor, as it might cause confusion
